src/Mod/TechDraw/Gui/QGCustomText.cpp

```diff
@@ -185,6 +191,32 @@ void QGCustomText::paint ( QPainter * painter, const QStyleOptionGraphicsItem *
     QGraphicsTextItem::paint (painter, &myOption, widget);
 }
 
+QRectF QGCustomText::boundingRect() const
+{
+    if (toPlainText().isEmpty()) {
+        return QRectF();
+    }
+
+    if (tightBounding) {
+        return tightBoundingRect();
+    }
+
+    return QGraphicsTextItem::boundingRect();
+}
+
+QRectF QGCustomText::tightBoundingRect() const
+{
+    QFontMetrics qfm(font());
+    QRectF result = QGraphicsTextItem::boundingRect();
+    QRectF tight = qfm.tightBoundingRect(toPlainText());
+    qreal x_adj = (result.width() - tight.width())/4.0;
+    qreal y_adj = (result.height() - tight.height())/4.0;
+
+    result.adjust(x_adj, 2*y_adj, -x_adj, -y_adj);
```

Comment:
Why `2*y_adj` here?

Reply:
The difference between the "loose" bounding box and the "tight" bounding box is divided into 4 parts, and in principle all of the adjustments should be 2*adj (i.e. 2/4 = 1/2).  However, this is too tight, because the "tight" Qt box touches the characters, and the original bounding box is not exactly vertically centered, so I made the bounding box halfway smaller, by chomping it smaller with 1*adj steps.  However, due to some Qt quirk there is a lot of empty extra space above the font, so I simply added 2*y_adj to eat up the ugly extra space.  Note that it never goes over 100 % (i.e. over 2*adj).

Admittedly, the commenting was lackluster, and the code was not very transparent (the underbox.Height()/4 was an adjustment to fix the difference between bounding boxes).  I made an addition to the pull request to for better commented and more general adjustment calculations.  Good comments, I see now that I was too hasty with the original coding, it's not enough if it works, it must be understandable too!